### DISCUSSION TITLE
Update time

### DISCUSSION THREAD

**Discussion by BrunoTeixeira1996** [gokrazy/gokrazy#258] [Q&A] [answered]
Hey everyone,

I've got a cronjob that's set to run daily at 18:00, but with the recent switch to summer time in my country, the schedule's been thrown off by an hour. Now it's running an hour later than I want it to.

I'm using a gokrazy instance, and I'm wondering how I can sync up the schedule with the time change automatically. Ideally, I'd like it to adjust itself again when we switch back to standard time and lose that extra hour.

Any ideas on how to tackle this?

**Comment by stapelberg** [comment, marked as answer]
When you say “cronjob”, do you mean a program running on gokrazy?

The gokrazy packer [copies /etc/localtime from the host to the instance](https://github.com/gokrazy/tools/blob/main/internal/packer/localtime.go), so as long as you create the instance on a computer with the desired time zone configured, the default time zone on the gokrazy instance should match what you want.

That said, you can always make your Go programs work in any environment (regardless of its default timezone) and use the desired time zone explicitly. For example, to schedule tasks in Zürich time, I use:

```go
        loc, err := time.LoadLocation("Europe/Zurich")
        if err != nil {
                panic(err)
        }
```

And then, when I need the current time, I use `time.Now().In(loc)`

**Reply by BrunoTeixeira1996** [reply]
Yes the cronjob is a go program running on gokrazy that sleeps until the correct time as come

Ive created the gokrazy instance on my machine but the time is different maybe because now its +1 hour since I created the gokrazy instance?

**Reply by stapelberg** [reply]
No, daylight saving time should be no problem. 

You can copy the /etc/localtime from your gokrazy instance and compare it with the files in /usr/share/zoneinfo on your original build machine to see which time zone file you have. Maybe attach it to this discussion.

**Reply by BrunoTeixeira1996** [reply]
On gokrazy I have the following
```console
$ /tmp/breakglass341682711 # cat /etc/localtime 
TZif2UTCTZif2UTC
UTC0
```

On my build machine I have the following
```console
$ timedatectl
               Local time: Mon 2024-04-15 09:31:33 WEST
           Universal time: Mon 2024-04-15 08:31:33 UTC
                 RTC time: Mon 2024-04-15 08:31:33
                Time zone: Europe/Lisbon (WEST, +0100)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

So I think the packer did not copied the correct time, however Ill force my app to use my choosen time zone

**Reply by BrunoTeixeira1996** [reply]
I've added `_ "time/tzdata"` in the header of the `main.go` file and looks like that works

**Reply by stapelberg** [reply]
> On my build machine I have the following

Can you show the output of `ls -l /etc/localtime`? What timedatectl displays is nice, but the file level is where the gokrazy packer picks up the time zone.

**Reply by BrunoTeixeira1996** [reply]
You mean this? @stapelberg 

```console
$ ls -l /etc/localtime
lrwxrwxrwx 1 root root 33 Sep 11  2023 /etc/localtime -> /usr/share/zoneinfo/Europe/Lisbon
```

**Reply by stapelberg** [reply]
Yeah. So, are you saying that when you build your instance on that machine, the gokrazy packer does not include  /usr/share/zoneinfo/Europe/Lisbon as /etc/localtime? Can you provide the output of the following commands please?

```
gok overwrite --root /tmp/root.squashfs
cd /tmp
unsquashfs root.squashfs etc/localtime
ls -l squashfs-root/etc/localtime
cat squashfs-root/etc/localtime
```

**Reply by BrunoTeixeira1996** [reply]
Ohhhhhhhh i am so dumb hehehe, I was checking the wrong machine.
The machine that builds the environment is in fact using the wrong time zone ... so packer is doing everything just fine!

**Reply by stapelberg** [reply]
Alright, I’m glad we could get to the bottom of this :)

**Reply by BrunoTeixeira1996** [reply]
Thanks and sorry for wasting your time
